akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -628,6 +822,15 @@ final class ClusterReceptionist(pubSubMediator: ActorRef, settings: ClusterRecep
   val virtualNodesFactor = 10
   var consistentHash: ConsistentHash[Address] = ConsistentHash(nodes, virtualNodesFactor)
 
+  var clientInteractions = HashMap.empty[ActorRef, DeadlineFailureDetector]
+  var clientsPublished = HashSet.empty[ActorRef]
```

Comment:
mutable fields + mutable datatypes = sad panda
Can it be immutable datastructures instead?